src/cogent3/app/__init__.py

```diff
@@ -18,39 +18,23 @@
 
 def _get_app_attr(name, obj, mod, is_composable):
     """returns app details for display"""
-    _input_types = getattr(obj, "_input_types", [])
-    _output_types = getattr(obj, "_output_types", [])
-    _data_types = getattr(obj, "_data_types", [])
 
-    if _input_types:
-        if type(_input_types) == str:
-            _input_types = [_input_types]
-        in_type = [{None: ""}.get(e, e) for e in _input_types]
-    else:
-        in_type = []
+    _types = {"_input_types": [], "_output_types": [], "_data_types": []}
```

Comment:
given these are now dictionary keys, no need for the leading underscore, i.e. `_types = {"input_types": [], "output_types": [], "data_types": []}